level: read objects and static objects in TRX

lahm86 opened the pull request:
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

This moves object and static object reading to TRX, and tidies up related injections. For now, static sprites remain in the individual level modules, but I will follow this up with a separate PR as I have an idea for allowing sprites and statics to effectively have the same slot (ref #2311).

This also interprets static object flags (collidable and visible) directly on read to improve usage readability. I don't think there are any invisible static meshes in the original levels, but there are test levels [here](https://github.com/MontyTRC89/Tomb-Editor/pull/913) with the different types. Testing otherwise should be straight-forward, it will be immediately obvious if objects aren't loading properly.

lahm86 committed:
level: read objects in TRX

This moves object reading into TRX. TR1 injections continue to read
objects manually as it handles omitted mesh data there where we want
to retain existing information.

lahm86 committed:
tr1/types: rename static object properties

This renames TR1 static object properties to match TR2.

lahm86 committed:
objects: move STATIC_INFO definitions to TRX

This moves the STATIC_INFO struct to TRX along with the ID enum. Both
games have the same number of types.

lahm86 committed:
level: read static objects in TRX

This moves static object reading into TRX.

lahm86 committed:
level: interpret static object flags on load

This expands static object flags into relevant collidable and visible
bools on load to simplify usage.

rr- commented on src/libtrx/include/libtrx/game/objects/ids.h:
```diff
@@ -6,3 +6,59 @@ typedef enum {
+    STATIC_DEBRIS_9       = 49,
+    STATIC_NUMBER_OF      = 50,
+    // clang-format on
+} GAME_STATIC_ID;
```
I think we can safely get rid of this enum type tbh and introduce MAX_STATICS instead.  WDYT?

lahm86 replied:
Good call, the types are never referenced so it makes sense to remove. Done.

lahm86 committed:
objects: remove static object enum

Static object IDs are not referenced as they are all treated the same,
so this removes the enum and adds a const limit for count.
